File: e2e/ts/main.js
const h = require('hanzoai');
(async () => {
  const base = process.env.HANZO_BASE_URL;
  const cfg = new h.Configuration({ basePath: base });
  let fail = 0;
  try {
    const r = await new h.ModelsApi(cfg).getModels();
    const n = (r.data && r.data.data) ? r.data.data.length : 0;
    console.log(`  ok  GET /v1/models  ${r.status}, ${n} models`);
  } catch (e) { console.log('  FAIL models:', e.message.slice(0,70)); fail++; }
  try {
    await new h.EngineApi(cfg).engineStatus();
    console.log('  FAIL engine: unauthenticated call reported success'); fail++;
  } catch (e) {
    const code = e.response ? e.response.status : e.code;
    console.log(`  ok  GET /v1/engine/status refused: ${code}`);
  }
  console.log(fail === 0 ? 'PASS typescript' : `FAIL typescript (${fail})`);
  process.exit(fail === 0 ? 0 : 1);
})();

File: e2e/rs/src/main.rs
use hanzo_client::apis::configuration::Configuration;

#[tokio::main]
async fn main() {
    let mut cfg = Configuration::new();
    cfg.base_path = std::env::var("HANZO_BASE_URL").unwrap();
    let mut fail = 0;

    match hanzo_client::apis::ai_api::get_models(&cfg).await {
        Ok(_) => println!("  ok  GET /v1/models  decoded"),
        Err(e) => { println!("  FAIL models: {e}"); fail += 1; }
    }
    match hanzo_client::apis::engine_api::engine_status(&cfg).await {
        Ok(_) => { println!("  FAIL engine: unauthenticated call reported success"); fail += 1; }
        Err(e) => println!("  ok  GET /v1/engine/status refused: {}", &format!("{e}")[..40.min(format!("{e}").len())]),
    }
    println!("{}", if fail == 0 { "PASS rust" } else { "FAIL rust" });
    std::process::exit(if fail == 0 { 0 } else { 1 });
}
